File: C#2/Homeworks/01-Arrays/05MaximalIncreasingSequence/MaximalIncreasingSequence.cs
using System;
/*
Write a program that finds the length of the maximal increasing sequence in an array of N integers.

Input
•On the first line you will receive the number N
•On the next N lines the numbers of the array will be given

Output
•Print the length of the maximal increasing sequence

Constraints
•1 <= N <= 1024
•Time limit: 0.1s
•Memory limit: 16MB
 */


class MaximalIncreasingSequence
{
   static void Main()
   {
       int n = int.Parse(Console.ReadLine());

       int count = 1;
       int maxIncreasingCount = 1;

       int[] arr = new int[n];
       for (int i = 0; i < n; i++)
       {
           arr[i] = int.Parse(Console.ReadLine());
       }

       for (int i = 0; i < n-1; i++)
       {
           if (arr[i] < arr[i + 1])
           {
               count++;
               if (count > maxIncreasingCount)
               {
                   maxIncreasingCount = count;
               }
           }
           else
           {
               count = 1;
           }
       }
       Console.WriteLine(maxIncreasingCount);
   }
}

File: JS/HW/Arrays/maximal-increasing-sequence.js
//Write a program that finds the length of the maximal increasing sequence in an array of N integers.
// On the first line you will receive the number N
// On the next N lines the numbers of the array will be given

function maximalIncreasingSequence(args){
    let n = +args[0],
        numbersArr = [],
        countIncreasing = 1,
        maxIncreasingCount = 1;

    for(let i = 1; i < args.length; i+= 1){
        numbersArr.push(+args[i]);
    }
    
    for(let i = 0; i < numbersArr.length; i+= 1){
        if(numbersArr[i+1] > numbersArr[i]){
            countIncreasing+=1;
            if(countIncreasing > maxIncreasingCount){
                maxIncreasingCount = countIncreasing;
            }
        }else
        {
            countIncreasing=1;
        }
    }
    console.log(maxIncreasingCount);
}

maximalIncreasingSequence(['8', '7', '3', '2', '3', '4', '2', '2', '4']);
maximalIncreasingSequence(['8', '7', '3', '2', '3', '4', '5', '6', '4']);